services/web/server/src/simcore_service_webserver/diagnostics_monitoring.py

```diff
@@ -32,14 +32,15 @@ def get_collector_registry(app: web.Application) -> CollectorRegistry:
 async def metrics_handler(request: web.Request):
     registry = get_collector_registry(request.app)
 
-    with concurrent.futures.ThreadPoolExecutor() as pool:
-        # NOTE: Cannot use ProcessPoolExecutor because registry is not pickable
-        result = await request.loop.run_in_executor(
-            pool, prometheus_client.generate_latest, registry
-        )
-        response = web.Response(body=result)
-        response.content_type = CONTENT_TYPE_LATEST
-        return response
+    # NOTE: Cannot use ProcessPoolExecutor because registry is not pickable
+    result = await request.loop.run_in_executor(
+        get_shared_thread_pool(request.app),
```

Comment:
did you check what happens when you pass None as the pool in run_in_executor? It should also use a threadpool. but is it also blocking? can you please comment on this?

Reply:
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor

If I understand well, there is a default executor. so maybe it's not necessary to use this whole infrastructure when using a ThreadPool right? only for ProcessPool?

Reply:
Apparently passing None, equals to using the default thread pool.
Apparently it looks like it should be safe to use the default system executor as there is no closing after it is being created. [source code here](https://github.com/python/cpython/blob/35b773accb41f09e40bf17bfaa5f0bc80796a26c/Lib/asyncio/base_events.py#L799-L813)